metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/EnrichmentService.java

```diff
@@ -74,43 +54,12 @@ public EnrichmentService(EnrichmentDao enrichmentDao) {
    */
   public List<EnrichmentResultBaseWrapper> enrichByEnrichmentSearchValues(
       List<SearchValue> searchValues) {
-    final List<EnrichmentResultBaseWrapper> enrichmentBases = new ArrayList<>();
-    try {
-      for (SearchValue searchValue : searchValues) {
-        findEnrichmentEntitiesBySearchValue(enrichmentBases, searchValue);
-      }
-    } catch (RuntimeException e) {
-      LOGGER.warn("Unable to retrieve entity from tag", e);
-    }
-    return enrichmentBases;
-  }
-
-  private void findEnrichmentEntitiesBySearchValue(
-      List<EnrichmentResultBaseWrapper> enrichmentBases, SearchValue searchValue) {
-    final String value = searchValue.getValue().toLowerCase(Locale.US);
-    if (!StringUtils.isBlank(value)) {
-      final List<EntityType> entityTypes = searchValue.getEntityTypes();
-      //Language has to be a valid 2 or 3 code, otherwise we do not use it
-      final String inputValueLanguage = searchValue.getLanguage();
-      final String language;
-      if (inputValueLanguage != null && inputValueLanguage.length() == 3) {
-        language = ALL_3CODE_TO_2CODE_LANGUAGES.get(inputValueLanguage.toLowerCase(Locale.US));
-      } else if (inputValueLanguage != null && inputValueLanguage.length() == 2) {
-        language = ALL_2CODE_LANGUAGES.contains(inputValueLanguage) ? inputValueLanguage : null;
-      } else {
-        language = null;
-      }
+    Set<SearchTerm> searchTerms = searchValues.stream().map(search -> new SearchTermType(search.getValue(), LanguageCodes
+        .convert(search.getLanguage()), search.getEntityTypes())).collect(Collectors.toSet());
```

Comment:
So, in conjunction with another review comment, the language conversion should go inside the resolver.

Reply:
Done